test/tests/functional/pbs_server_periodic_hook.py

```diff
@@ -93,18 +93,18 @@ def check_next_occurances(self, count, freq,
                                         interval=(intr + 1),
                                         starttime=search_after)
             time_logged = self.get_timestamp(msg[1])
-            self.assertFalse((time_logged - time_expected) > 1)
+            self.assertFalse((time_logged - time_expected) > freq)
```

Comment:
I do not think the issue is because of slower machines, the issue is the wrong expectation in the first iteration.

time_expected is set to an arbitrary value before we enter the loop. We do not know when the periodic hook ran before we called this function. So, we are assuming that it should run before now + freq, which is correct, but we should not be testing for the time difference as the reference point (time now) itself is an assumed one.
We should test for the time difference only from the second iteration after setting time_expected in the first iteration, and in that case, the difference should not be greater than 1.